source/_posts/2017-06-14-Ansible-via-python.markdown

```diff
@@ -0,0 +1,198 @@
+two.example.com
+three.example.com
+```
+
+## Ansible Playbook
+
+Playbook is the way to direct systems, which is particularly powerful, compact
+and easy to read and write. As we said configuration multi-system management is
+formatted in YAML language. Ansible playbook consists of `plays`, which
+contain `hosts` we would like to manage, and `tasks` we want to perform.
+
+```yaml
+---
+- hosts: webservers
+  vars:
+    http_port: 80
+    max_clients: 200
+  remote_user: root
+  tasks:
+  - name: ensure apache is at the latest version
+    yum: name=httpd state=latest
+  - name: write the apache config file
+    template: src=/srv/httpd.j2 dest=/etc/httpd.conf
+    notify:
+    - restart apache
+  - name: ensure apache is running (and enable it at boot)
+    service: name=httpd state=started enabled=yes
+  handlers:
+    - name: restart apache
+      service: name=httpd state=restarted
+```
+
+### Ansible for Embedded Linux
+
+> Note: This paragraph is relevant to Yocto build system
+
+There is possibility to need build image with custom Linux-based system for
+embedded with Ansible, using complete development environment, with tools,
+metadata and documentation, named Yocto. In addition, we would like to run
+ansible-playbook via python. It seems to be hard to implement. Nothing more
+simple! It is needed to add recipe to image with ansible from
+[Python Ansible package](https://github.com/OverC/meta-overc/blob/master/meta-cube/recipes-devtools/python/python-ansible_2.1.1.0.bb)
+
+#### Additional information
+
+For more information go to [Ansible Documentation](http://docs.ansible.com/ansible/)
+
+## Python API
+
+Python API is very powerful, so we can manage and run ansible-playbook from
+python level, there is possibility to controll nodes, write various plugins,
+extend Ansible to respond to various python events and plug in inventory data
+from external data sources.
+
+> Note: There is a permament structure to build python program which operates
+ansible commands:
+
+First of all we have to import some modules needed to run ansible in python.
+
+- Let's describe some of them:
+
+  - `json` module to convert output to json format
+  - `ansible` module to manage e.g. inventory or plays
+  - `TaskQueueManager` is responsible for loading the play strategy plugin,
+  which dispatches the Play's tasks to hosts
+  - `CallbackBase` module - base ansible callback class, that does nothing
+  here, but new callbacks, which inherits from CallbackBase class and
+  override methods, can execute custom actions
+
+```python
+import json
+from collections import namedtuple
+from ansible.parsing.dataloader import DataLoader
+from ansible.vars import VariableManager
+from ansible.inventory import Inventory
+from ansible.playbook.play import Play
+from ansible.executor.task_queue_manager import TaskQueueManager
+from ansible.plugins.callback import CallbackBase
+```
+
+- ResultCallback which inherits from `CallbackBase` and manage the output of
```

Comment:
@Macias118 I don't feel you changed a lot in recent commit. Am I missing something ? Please elaborate on each line with at least 2 sentences. Please give reader feeling that you understand this code.